### ISSUE TITLE
Invalid enum names not properly handled (e.g.: "mro")

### ISSUE THREAD

**Issue by DavideWalder** [koxudaxi/datamodel-code-generator#873]
**Describe the bug**
Enum member names can't be `mro`, since this clashes with the method resolution order method.

**To Reproduce**

Example schema:
```json
{
  "type": "array",
  "items": {
    "type": "object",
    "properties": {
      "enum_with_invalid_value": {
        "type": "string",
        "enum": ["mro"]
      }
    }
  }
}
```

Used commandline:
```sh
datamodel-codegen  --input schema.json  --input-file-type jsonschema --output model.py
python -c "import model" # ValueError: Invalid enum member name: mro
```

**Expected behavior**
The name should be adapted. According to [PEP-8](https://peps.python.org/pep-0008/#function-and-method-arguments), the preferred solution is to suffix with an underscore (`"mro_"` instead of `"mro"`). Since this is only the enum name it should not have any impact on parsing.

**Version:**
 - OS: MacOS
 - Python version: 3.10.4
 - datamodel-code-generator version: 0.13.1

**Comment by koxudaxi** [comment]
@DavideWalder 
I have released the new version as `0.13.5`
Thank you very much!!

**Comment by DavideWalder** [comment]
@koxudaxi
I'm testing with 0.17.0 and the issue is still there (same for 0.14.0)

**Comment by DavideWalder** [comment]
Found the issue:

input:
```yaml
openapi: 3.0.3
components:
  schemas:
    TheEnum:
      type: string
      enum:
        - MRO
        - mro
```

output:
```python
# generated by datamodel-codegen:
#   filename:  datasync.yaml
#   timestamp: 2023-01-31T13:27:19+00:00

from __future__ import annotations

from enum import Enum


class TheEnum(Enum):
    mro = 'MRO'
    mro_ = 'mro'
```

The underscore is suffixed only when the enum value is lowercase

**Comment by koxudaxi** [comment]
Thank you for explaining the problem.
I will soon change the result to the one below.
```python
class TheEnum(Enum):
    mro_ = 'MRO'
    mro_1 = 'mro'
```

**Comment by DavideWalder** [comment]
Also, this brings up the question of what to do when two entries differ by casing like `NAME` and `name`. Probably a clearer solution would be to maintain the casing, as:

```python
class TheEnum(Enum):
    MRO_ = 'MRO'
    mro_ = 'mro'
```

This would also avoid the issue where changing the order from
```yaml
openapi: 3.0.3
components:
  schemas:
    TheEnum:
      type: string
      enum:
        - MRO
        - mro
```
```yaml
openapi: 3.0.3
components:
  schemas:
    TheEnum:
      type: string
      enum:
        - mro
        - MRO
```
would change the enum names.

I see that keeping the casing would be a non-backward compatible change though

**Comment by koxudaxi** [comment]
@DavideWalder 
Good point.
I want to avoid changing current behavior.
I reccomend to use `--snake-case-field` or `--capitalise-enum-members`

without options
```python
class InvalidEnum(Enum):
    MRO = 'MRO'
    mro_ = 'mro'
```

 `--snake-case-field`
 ```python
 class InvalidEnum(Enum):
    mro_1 = 'MRO'
    mro_ = 'mro'
 ```
 
 `--capitalise-enum-members`
```python
class InvalidEnum(Enum):
    MRO = 'MRO'
    MRO_ = 'mro'
 ```

https://github.com/koxudaxi/datamodel-code-generator/pull/1069/files#diff-440ee6529a7880f5b3f955ee7186af0b603ba307adc6991ea004642878e55182R15-R16

**Comment by koxudaxi** [comment]
@DavideWalder 
I have released the fix version `0.17.1`
Thank you very much!!
